extensions/workloads/template.go

```diff
@@ -118,3 +119,22 @@ func NewJobTemplate(jobName string, namespace string) *batchv1.Job {
 	}
 
 }
+
+// NewPodTemplateWithSecretEnvironmentVariable is a constructor that returns pod template spec with envFrom option for workload creations
+func NewPodTemplateWithSecretEnvironmentVariable(secretName string) corev1.PodTemplateSpec {
+	containerName := namegen.AppendRandomString("testcontainer")
```

Comment:
this template always uses nginx image. it might be best if we get the containerName and containerImages/containerName from user as input.